### PR TITLE
restic backup --tag integration test

### PR DESCRIPTION
**Pull request by middelink**
Make sure the backup command does indeed add the required tags when asked.

### CODE REVIEW

File: src/cmds/restic/integration_test.go [hunk 1]
```diff
@@ -160,6 +161,33 @@ func testRunFind(t testing.TB, gopts GlobalOptions, pattern string) []string {
...
+		globalOptions.stdout = os.Stdout
+		globalOptions.JSON = gopts.JSON
+	}()
+
+	opts := SnapshotOptions{}
+
+	OK(t, runSnapshots(opts, globalOptions, []string{}))
+
+	snapshots := []Snapshot{}
+	OK(t, json.Unmarshal(buf.Bytes(), &snapshots))
```
Comment (fd0) [inline src/cmds/restic/integration_test.go]: Nice idea, parsing the JSON. I like it :)

File: src/cmds/restic/integration_test.go [hunk 2]
```diff
@@ -602,6 +630,31 @@ func TestIncrementalBackup(t *testing.T) {
...
+	withTestEnvironment(t, func(env *testEnvironment, gopts GlobalOptions) {
+		datafile := filepath.Join("testdata", "backup-data.tar.gz")
+		testRunInit(t, gopts)
+		SetupTarTestFixture(t, env.testdata, datafile)
+
+		opts := BackupOptions{}
+
+		testRunBackup(t, []string{env.testdata}, opts, gopts)
+		testRunCheck(t, gopts)
+		newest, _ := testRunSnapshots(t, gopts)
```
Comment (fd0) [inline src/cmds/restic/integration_test.go]: Why do you return the snapshot map if you're not going to use it here (the second return value of `testRunSnapshots()`)?

**Review by fd0 — CHANGES_REQUESTED** [review]
Looks already very good, just a minor thing :)

File: src/cmds/restic/integration_test.go (hunk 2 above)
Comment (middelink) [inline src/cmds/restic/integration_test.go]: I wanted to be generous to other potential testcases which might have a need to get all snapshots. I can change it's signature and leave it to others to add it back if you like?

Comment (fd0) [inline src/cmds/restic/integration_test.go]: Ok, good point, on a second thought that can be quite useful. Looked like old code for me when I first read it, so leave it in.

**Review by fd0 — APPROVED** [review]
Thanks for your work!

**Pull request merged**